PhotoGalleryXamarin/Fragments/PhotoGalleryFragment.cs

```diff
@@ -63,6 +66,34 @@ public override View OnCreateView(LayoutInflater inflater, ViewGroup container,
 
             return view;
         }
+
+        public override void OnCreateOptionsMenu(IMenu menu, MenuInflater inflater)
+        {
+            base.OnCreateOptionsMenu(menu, inflater);
+
+            inflater.Inflate(Resource.Menu.fragment_photo_gallery, menu);
+
+            var searchItem = menu.FindItem(Resource.Id.menu_item_search);
+            _searchView = (Android.Support.V7.Widget.SearchView)searchItem.ActionView;
+
+            _searchView.QueryTextSubmit += QueryTextSubmitted;
+            _searchView.QueryTextChange += QueryTextChanged;
+            _searchView.SetOnSearchClickListener(_searchClickListener);
```

Comment:
Let's create the listener here, instead of using a field, and make the class take the action in constructor instead of setter